sway-core/src/optimize.rs

```diff
@@ -1113,6 +1113,7 @@ impl FnCompiler {
                 .iter()
                 .map(|(name, expr)| TypedFunctionParameter {
                     name: name.clone(),
+                    is_mutable: false,
```

Comment:
This will need to be consistent with the original function.  I hate this code so much now, often I'm wishing we just inlined the callee here rather than later because of things like this (originally the amount of work to do one or the other was 50/50 and so I just did it this way semi-arbitrarily).

So, if we call a function with a mutable self in Sway, then recreate that function here, and then inline it later, does it still work?

Looking at the tests below, I guess it must. :slightly_smiling_face: Maybe I'm missing something here, but I'd assume just setting mutable to false here would break things.

Reply:
`mut` for `self` only matters in type checking which is why this works. That being said, it is odd and I cringed writing it. Mutability here should probably be propagated to arguments in IR and validated in the verifier.

Reply:
OK.  We need `Vec` more than we need this to be perfect. :smile: